planner/mapper/mapper.go

```diff
@@ -928,6 +906,131 @@ sourceLoop:
 	return newFields, nil
 }
 
+// constructDummyJoin constructs a valid empty join with no requested fields.
+func constructDummyJoin(
```

Comment:
suggestion: I think we should avoid using the work `Dummy` as it feels more like something you'd use just for testing purposes. In this case this is a function that is used in a critical path. Maybe `constructEmptyJoin` would be more appropriate.

Reply:
Yeah good shout - I'll change it

- [ ] Rename Dummy func